### PR TITLE
Clickhouse support

### PR DESCRIPTION
**Pull request by strug**
Use pr https://github.com/flyway/flyway/pull/2981 - update for new scheme implementation. Fix some problem with distributed logic, when more one instance application.

### CODE REVIEW

**Comment by GordeevArt** [comment]
In flyway replicated table creation zookeeper path should be "/clickhouse/{installation}/{cluster}/tables/{shard}/{database}/{table}" instead of "/clickhouse/tables/{shard}/" + tableName. It prevents conficts with mutliple clickhouse installations with flyway in one zookeeper.

**Comment by den-crane** [comment]
> In flyway replicated table creation zookeeper path should be

should be???
99% of CH users don't have macros {cluster}
100% of CH users don't have macros {installation}
It simply won't work.

**Comment by GordeevArt** [comment]
> > In flyway replicated table creation zookeeper path should be
> 
> should be??? 99% of CH users don't have macros {cluster} 100% of CH users don't have macros {installation} It simply won't work.

Maybe add **zookeeper.path** parameter?

**Comment by den-crane** [comment]
BTW it's a Russian С not С in the PR name "Сlickhouse support"
That's confusing , I copy paste Сlickhouse word and was unable to find it.

**Comment by johnny** [comment]
What is the status of this PR and the chances of it getting merged? We'd like to use flyway for clickhouse.

**Comment by harshvikramsingh** [comment]
Hi, we are also waiting for this change. Can this be please merged if no more issues?

File: flyway-commandline/pom.xml [hunk 1]
```diff
@@ -185,6 +185,11 @@
             <artifactId>lombok</artifactId>
         </dependency>
 
+        <dependency>
```
Comment (TheHett) [inline flyway-commandline/pom.xml:188]: this driver is obsolete

**Comment by Barry-RG** [comment]
Clickhouse has been added via subsequent PRs

**Pull request closed without merging**